### PR TITLE
Add version and versionUrl props to Footer component

### PR DESCRIPTION
**Pull request by DanielBaird**
version prop to include display of version number/string in the footer 
also, adds .env file to make library version available (so Footer demo can use it as the default version)
also, establishes new convention for changelog: bold component names

### CODE REVIEW

**Comment by DanielBaird** [comment]
Updated as discussed.  I also added some edge case tests that pushed the coverage up to 100% :)

File: src/layout/Footer.test.js [hunk 1]
```diff
@@ -11,3 +11,53 @@ test('shows current year', () => {
...
+  const footer = shallow(<Footer version={verString} />)
+  expect(footer.contains(verString)).toBe(true)
+})
+
+test('handles boolean version prop values', () => {
+  let footer = shallow(<Footer version={false} />)
+  expect(footer.find('.footer-version')).toHaveLength(0)
+
+  footer = shallow(<Footer version={true} />)
+  expect(footer.find('.footer-version')).toHaveLength(1)
```
Comment (davidjb) [inline src/layout/Footer.test.js]: Added these `expect` clauses so we check that the footer version is being rendered when it should and shouldn't.  Previously, the tests just tested if rendering worked but not the outcome.

Comment (DanielBaird) [reply src/layout/Footer.test.js]: I think I need to be more careful about refreshing my pull req conversations -- twice now I've noticed a code update but missed the related comment.

Yes I originally wrote those tests to just confirm no crash; passing a boolean `false` as the prop value is technically allowed but not useful.  I think there's a valid case where passing `true` (the code reformatting removes the explicit boolean `true`, so it's just prop-exists-true) when there's no env var set will result in no version number appearing. I'll add an issue for that so i remember to come back around and fix it.
